frontend/src/Contexts/product-context.js

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext, useReducer } from "react";
+import { productsByCategory } from "../Reducers/products";
+import { compose, sortedPrice } from "../Utils/products";
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productsByCategory, {
+    loading: false,
+    product: [],
+    success: false,
+    error: false,
+    sort_price: "normal",
+    maxPrice: 5000,
+  });
```

Comment:
suggestion: U can extract this values in a separated variable like initial value or something